lib/flash.robot

```diff
@@ -185,3 +185,11 @@ Compare Write Protection Ranges
     IF    ${set_length}!=${length}
         FAIL    Declared and currently set protection lengths are not the same
     END
+
+Read Firmware
+    [Documentation]    Read platform firmware to file specified in the argument.
+    [Arguments]    ${file}    ${flags}=""
+    RteCtrl Power Off
+    Sleep    2s
+    SSHLibrary.Execute Command    flashrom -p linux_spi:dev=/dev/spidev1.0,spispeed=16000 -r /tmp/coreboot.rom ${flags}
```

Comment:
Unfortunately, it will work only in a  very specific case, for APU platforms setup with RTE rev v1.0.0 or lower.
For different RTE revision, or different platform, it will not work.

Only after implementing: https://github.com/Dasharo/open-source-firmware-validation/issues/257 we will be able to have more generic read/write keywords in this repo I am afraid.

Please link to this issue and detect the platform type. If it is different than any of the apu, fail with a message that this kwd is not supported for this platform yet.

Reply:
Is this what you meant by linking the issue?: 90f743f7b6f3b46b360c57f2725f60a7854b7013